htdocs/core/triggers/interface_20_modWorkflow_WorkflowManager.class.php

```diff
@@ -315,6 +316,39 @@ public function runTrigger($action, $object, User $user, Translate $langs, Conf
 			}
 		}
 
+		// classify project as closed when progress of all tasks is 100%
+		if ($action == 'TASK_MODIFY') {
+			//
+			if (!empty($conf->projet->enabled) && !empty($conf->global->WORKFLOW_PROJECT_CLASSIFY_CLOSED_WHEN_ALL_TASKS_DONE))
```

Comment:
Should I also test whether the project is validated?

Reply:
I think so. A draft must stay a draft.